### ISSUE TITLE
Documentation class reference static check refuses setter/getter references despite working just fine

### ISSUE THREAD

**Issue by Mickeon** [godotengine/godot#87032]
### Tested versions

4.3 dev (as of writing, master)

### System information

Does not matter

### Issue description

As discovered by https://github.com/godotengine/godot/pull/87030#issuecomment-1884447779

The static check for the documentation will fail when directly referencing a property's setter or getter method.

![image](https://github.com/godotengine/godot/assets/66727710/bb39d23d-9520-42fc-b7fc-8e1c3be40f2f)

This seems appropriate, since there's no distinct description to send users to, as both setters and getters are clumped together with their respective property in the documentation.

Despite this, the built-in class reference seems to support this just fine.

<img src=https://i.gyazo.com/473839b2db9bb5318ef56c25bf2768dd.gif width=30%>

Could it be that the **online** class reference generation does not support it, hence this error?

### Steps to reproduce

- Write some documentation referencing setter/getters of a built-in property. It can either be in the Godot repository, or through GDScript.
- Notice that setters and getters are appropriately detected as methods by the doc gen.
- Despite this, the static check would fail with them.

### Minimal reproduction project (MRP)

Not necessary

**Comment by YuriSizov** [comment]
Linking to something in online docs is just a matter of defining an anchor, it's trivial to add.

But I think we don't want to do that, because we should not use getters and setters over properties in descriptions. As a matter of principle, because at some point it was decided that properties are preferable.

**Comment by Mickeon** [comment]
Perhaps the error could instead be a warning, in this case? Or the error could be more explicit about this.
The way this issue was discovered is through an edge case. For GDExtension it makes more sense to reference the setter/getter overridden by the methods over the underlying property. At least, that's how I feel about it.

**Comment by Mickeon** [comment]
I don't think we _absolutely_ need an exception. I just believe that GDExtension developers would be more interested in seeing the setters and getters, since it is what they're directly influencing by overriding them. However they want to expose these properties is up for them to decide, if need to be, too. But it's not incredibly necessary.

**Comment by YuriSizov** [comment]
@Mickeon Could you give a concrete example? I'm not sure I fully understand the context here.

If it's for the GDExtension related classes which we document in our class reference, then it should follow our general rules and use properties. We also don't normally allow to override public methods like setters and getters directly. You instead override their virtual counterparts. Which are documented separately.

If it's for 3rd party extensions using our documentation framework, then it may be open for discussion (outside of this issue, probably). But I would say that for the consistency's sake, if an extensions exposes a property and not just methods, then it should also follow our general rules and prefer properties in the docs.

**Comment by Mickeon** [comment]
No need of an example. I understand and agree with you.

**Comment by akien-mga** [comment]
In this case, properties are a language specific construct, so they don't exist in GDExtension. GDExtension exposes virtual methods to override for the setters and getters which are used in the engine code for a given core property.

See examples in https://github.com/godotengine/godot/pull/87030. `PhysicsDirectBodyState2DExtension` provides `_set`/`_get_center_of_mass`, but no property. The non-`Extension` counterpart, `PhysicsDirectBodyState2D`, has the `center_of_mass` property with its setter and getter being "hidden" to favor the property. Since GDExtension is lower level, you have to override the setter and getter, not the property itself (a property is just a setter and a getter in a ClassDB trenchcoat).

That being said, while I understand the slight inconvenience for documenting `Extension` classes, I think it's not a big deal and `_get_center_of_mass` can refer to the property in the non-`Extension` class, like it does in the current PR.
